pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -972,4 +1017,41 @@ public void onUpdate(String path, Map<String, String> data, Stat stat) {
         return dynamicConfigurationMap;
     }
 
-}
+    /**
+     * Create pending topic and on completion it picks the next one until processes all topics in
+     * {@link #pendingTopicLoadingQueue}.<br/>
+     * It also tries to acquire {@link #topicLoadRequestSemaphore} so throttle down newly incoming topics and release
+     * permit if it was successful to acquire it.
+     */
+    private void createPendingLoadTopic() {
+        Pair<String, CompletableFuture<Topic>> pendingTopic = pendingTopicLoadingQueue.poll();
+        if (pendingTopic != null) {
+            final String topic = pendingTopic.getLeft();
+            try {
+                checkTopicNsOwnership(topic);
+                CompletableFuture<Topic> future = new CompletableFuture<>();
+                final Semaphore topicLoadSemaphore = topicLoadRequestSemaphore.get();
+                final boolean acquiredPermit = topicLoadSemaphore.tryAcquire();
```

Comment:
actually with this approach compare to my previous [commit-406b0fc](https://github.com/yahoo/pulsar/pull/306/commits/406b0fc6b8efcc1451e163858a1e3f72860968fd) fails to provide throttling in certain usecases.
eg:
1. maxConcurrentTopicLoadRequest=10 and broker receives 21 requests so, 10 will be executed and rest of them will go to pending-queue
2. once first 10 completes it will try to execute pending-10 without any throttling. but at the same time if additional 31 requests come then 10 requests out of them will be served as semaphore is already released. so, now we are serving total 20 concurrent request (10(new requests) + 10(10 completed topic threads which invoked `createPendingLoadTopic`)
3. and this process go on which can increase concurrent-count and we are not doing real throttling.

- However, I am acquiring semaphore here so, at least those new additional coming 31 requests will be throttled. Still this approach doesn't throttle 100% in compare to [commit-406b0fc](https://github.com/yahoo/pulsar/pull/306/commits/406b0fc6b8efcc1451e163858a1e3f72860968fd)